Document/0x06i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -216,23 +216,119 @@ In swift 3, using xCode 8, one can set Active Compilation Conditions setting in
+* That `NSError` objects do not contain information that might leak any sensitive information.
+
+##### Static Analysis in Swift
+Here you can verify that:
+
+* The application uses a well-designed and unified scheme to handle errors.
+* The application doesn't expose sensitive information while handling errors in its UI or in its log-statements, but are still verbose enough to explain the issue to the user.
+* The throws are handled correctly: e.g. 
+* That any confidential information, such as keying material and/or authentication information is always wiped at the `defer` blocks in case of a high risk application.
+* That `try!` is only used with proper guarding up front, so it is programmatically verified that indeed no error can be thrown by the method that is called using `try!`.
 
 #### Dynamic Testing
 
--- TODO [Describe how to test for this issue "Testing Exception Handling" using static and dynamic analysis techniques. This can include everything from simply monitoring aspects of the app’s behavior to code injection, debugging, instrumentation, etc. ] --
+There are various methods for dynamic analysis:
 
-#### Remediation
+- Provide unexpected values to UI fields in the iOS application.
+- Test the custom url-schemes, pasteboard and other inter-app communication controls by providing values that are unexpected or could raise an exception.
+- Tamper the network communication and/or the files stored by the application.
+- In case of Objective-C, you can use Cycript to hook into methods and provide them with arguments that could possibly make the callee throw an exception.
+
+In most cases, the application should not crash, but instead, it should:
 
--- TODO [Describe the best practices that developers should follow to prevent this issue "Testing Exception Handling"] --
+- Recover from the error or get into a state in which it can inform the user that it is not able to continue.
+- If necessary, inform the user in an informative message to make him/her take appropriate action. The message itself should not leak sensitive information.
+- Not provide any information in logging mechanims used by the application.
+
+#### Remediation
+There are a few things a developer can do:
+- Ensure that the application use a well-designed and unified scheme to handle errors.
+- Make sure that all logging is removed or guarded as described in ["Testing for Debugging Code and Verbose Error Logging" for iOS]{TODO: whatlinkshouldbehere?}.
+- For Objective-C, in case of a high-risk application: create your own exception handler which cleans out any secret that should not be easily retrieved. The handler that can be set through `NSSetUncaughtExceptionHandler`.
+- When using Swift, make sure that you do not use `try!` unless you have made sure that there really cannot be any error in the method the throwing method that is being called.
+- When using Swift, make sure that the error does not propagate too far off.
```

Comment:
That is true, but i rather say something like "- When using Swift, make sure that the error does not propagate too far off through intermediate methods." is that good as well?